### PR TITLE
passkeys: check p256 signature

### PR DESCRIPTION
**Pull request by aramikm**
# Goal
The goal of this PR is check p256 Passkey signatures validations.

Closes #2036 

# Checklist
- [X] Chain spec updated
- [X] Tests added
- [X] Benchmarks updated
- [X] Weights updated

### CODE REVIEW

File: pallets/passkey/src/weights.rs [hunk 1]
```diff
@@ -44,20 +44,20 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
 	/// Proof: `System::Account` (`max_values`: None, `max_size`: Some(128), added: 2603, mode: `MaxEncodedLen`)
 	fn validate() -> Weight {
 		// Proof Size summary in bytes:
-		//  Measured:  `39`
+		//  Measured:  `52`
 		//  Estimated: `5078`
-		// Minimum execution time: 42_000_000 picoseconds.
-		Weight::from_parts(43_000_000, 5078)
+		// Minimum execution time: 988_000_000 picoseconds.
+		Weight::from_parts(1_002_000_000, 5078)
```
Comment (aramikm) [inline pallets/passkey/src/weights.rs:50]: This is a massive jump and worrying. We might be able to optimize it a bite but not a lot. In worst case we might need to move that p256 signature check into extrinsic itself

**Review by saraswatpuneet — COMMENTED** [review]
Overall looks good :+1:

File: pallets/passkey/src/lib.rs [hunk 2]
```diff
@@ -266,6 +242,73 @@ where
...
+
+		let verified = signature.verify(&signed_payload[..], &key);
+		if verified {
+			Ok(())
+		} else {
+			Err(Error::<T>::InvalidAccountSignature.into())
+		}
+	}
+}
+
```
Comment (saraswatpuneet) [inline pallets/passkey/src/lib.rs:311]: Nice , cleaner

File: pallets/passkey/src/lib.rs [hunk 3]
```diff
@@ -194,50 +206,14 @@ where
...
 }
 
 /// Passkey specific nonce check
 #[derive(Encode, Decode, Clone, TypeInfo)]
 #[scale_info(skip_type_params(T))]
-struct PasskeyNonce<T: Config>(pub PasskeyCall<T>);
+struct PasskeyNonceCheck<T: Config>(pub PasskeyCall<T>);
 
-impl<T: Config> PasskeyNonce<T>
+impl<T: Config> PasskeyNonceCheck<T>
```
Comment (enddynayn) [inline pallets/passkey/src/lib.rs:216]: Nice!

**Review by saraswatpuneet — APPROVED** [review]
Lgtm 👍🏽 nice work

**Review by enddynayn — APPROVED** [review]
Great! 👍

**Review by JoeCap08055 — APPROVED** [review]
nice work

**Pull request merged**